pkg/il/builder.go

```diff
@@ -0,0 +1,281 @@
+	f.op1(TResolveM, f.id(n))
+}
+
+// IPushBool appends the "ipush_b" instruction to the byte code.
+func (f *Builder) IPushBool(b bool) {
+	f.op1(APushB, boolToOpcode(b))
+}
+
+// IPushStr appends the "ipush_s" instruction to the byte code.
+func (f *Builder) IPushStr(s string) {
+	f.op1(APushS, f.id(s))
+}
+
+// IPushInt appends the "ipush_i" instruction to the byte code.
+func (f *Builder) IPushInt(i int64) {
+	a1, a2 := integerToOpcode(i)
+	f.op2(APushI, a1, a2)
+}
+
+// IPushDouble appends the "ipush_d" instruction to the byte code.
+func (f *Builder) IPushDouble(n float64) {
+	a1, a2 := doubleToOpcode(n)
+	f.op2(APushD, a1, a2)
+}
+
+// Xor appends the "xor" instruction to the byte code.
+func (f *Builder) Xor() {
+	f.op0(Xor)
+}
+
+// EQString appends the "eq_s" instruction to the byte code.
+func (f *Builder) EQString() {
+	f.op0(EqS)
+}
+
+// IEQString appends the "ieq_s" instruction to the byte code.
+func (f *Builder) IEQString(v string) {
+	f.op1(AEqS, f.id(v))
+}
+
+// EQBool appends the "eq_b" instruction to the byte code.
+func (f *Builder) EQBool() {
+	f.op0(EqB)
+}
+
+// IEQBool appends the "ieq_b" instruction to the byte code.
+func (f *Builder) IEQBool(v bool) {
+	f.op1(AEqB, boolToOpcode(v))
+}
+
+// EQInteger appends the "eq_i" instruction to the byte code.
+func (f *Builder) EQInteger() {
+	f.op0(EqI)
+}
+
+// IEQInteger appends the "eq_i" instruction to the byte code.
+func (f *Builder) IEQInteger(v int64) {
+	a1, a2 := integerToOpcode(v)
+	f.op2(AEqI, a1, a2)
+}
+
+// EQDouble appends the "eq_d" instruction to the byte code.
+func (f *Builder) EQDouble() {
+	f.op0(EqD)
+}
+
+// IEQDouble appends the "ieq_d" instruction to the byte code.
+func (f *Builder) IEQDouble(v float64) {
+	a1, a2 := doubleToOpcode(v)
+	f.op2(AEqD, a1, a2)
+}
+
+// Not appends the "not" instruction to the byte code.
+func (f *Builder) Not() {
+	f.op0(Not)
+}
+
+// Or appends the "or" instruction to the byte code.
+func (f *Builder) Or() {
+	f.op0(Or)
+}
+
+// And appends the "and" instruction to the byte code.
+func (f *Builder) And() {
+	f.op0(And)
+}
+
+// Lookup appends the "lookup" instruction to the byte code.
+func (f *Builder) Lookup() {
+	f.op0(Lookup)
+}
+
+// ILookup appends the "ilookup" instruction to the byte code.
+func (f *Builder) ILookup(v string) {
+	f.op1(ALookup, f.id(v))
+}
+
+// AllocateLabel allocates a new label value for use within the code.
+func (f *Builder) AllocateLabel() string {
+	l := fmt.Sprintf("L%d", len(f.labels))
+	f.labels[l] = 0
+	return l
+}
+
+// SetLabelPos puts the label position at the current bytecode point that builder is pointing at.
+// Panics if the label position was already set.
+func (f *Builder) SetLabelPos(label string) {
+	if f.labels[label] != 0 {
+		panic("interpreter.Builder: setting the label position twice.")
+	}
+	adr := uint32(len(f.body))
+	f.labels[label] = adr
+	if fixups, found := f.fixups[label]; found {
+		for _, fixup := range fixups {
+			f.body[fixup] = adr
+		}
+	}
```

Comment:
I don't think so. This is unlikely to be a big piece of data and  all of this goes out of scope pretty quickly.